controllers/rhmi/rhmi_controller.go

```diff
@@ -434,6 +465,191 @@ func (r *RHMIReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 	return retryRequeue, err
 }
 
+func (r *RHMIReconciler) createSilence(installation *rhmiv1alpha1.RHMI, ctx context.Context, rc *rest.Config) error {
+
+	var alertingNamespaces = map[string]string{
+		"openshift-monitoring": "alertmanager-main",
+		installation.Spec.NamespacePrefix + "middleware-monitoring-operator": "alertmanager-route",
+	}
+
+	for namespace, route := range alertingNamespaces {
+		for _, alert := range alertsToSilence {
+			exists, err := r.silenceExists(installation, namespace, route, log, rc, alert)
+			if err != nil {
+				log.Error("Error checking for silence", err)
+			}
+			if !exists {
+				r.silenceAlert(installation, namespace, route, log, rc, alert)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *RHMIReconciler) silenceAlert(installation *rhmiv1alpha1.RHMI, namespace string, route string, log l.Logger, rc *rest.Config, alertName string) error {
+
+	endpoint := "/api/v1/silences"
+	startsAt := time.Now().Format(time.RFC3339)
+	endsAt := time.Now().Add(time.Hour * 1).Format(time.RFC3339)
+
+	var silence = []byte(`{` +
+		`"endsAt": "` + endsAt + `",` +
+		`"startsAt": "` + startsAt + `",` +
+		`"matchers":[{"isRegex": false, "name": "alertname","value": "` + alertName + `"}],` +
+		`"Comment":"Silence alert due to uninstall",` +
+		`"createdBy":"Integreatly Operator"}`)
+
+	url, err := r.getURLFromRoute(route, namespace, rc)
+	if err != nil {
+		return fmt.Errorf("error getting route : %w", err)
+
+	}
+	url = url + endpoint
+	serviceAccount := r.getServiceAccount(installation)
+
+	token, err := r.getBearerToken(installation, serviceAccount)
```

Comment:
The token should be in the rest config instance that is attached to this reconciler: `r.restConfig.BearerToken`

Docs: https://pkg.go.dev/k8s.io/client-go/rest#Config

Reply:
This suggestion is implemented now. Marking as resolved.